src/js/game/hud/parts/base_toolbar.js

```diff
@@ -103,15 +103,25 @@ export class HUDBaseToolbar extends BaseHUDPart {
      * Cycles through all buildings
      */
     cycleBuildings() {
+        const visible = this.visibilityCondition();
+        if (!visible) {
+            return;
+        }
+
+        let process = false;
```

Comment:
Could you rename this to `anyBuildingMatched` or something similar?

Reply:
I changed the variable name to `newBuildingFound`.